### DISCUSSION TITLE
[Inline Tool] Mentions

### DISCUSSION THREAD

**Discussion by neSpecc** [codex-team/editor.js#1880] [Q&A]
Design: [zpl.io/aBN4oBp](//zpl.io/aBN4oBp)

Config `items`: 

```
Array of {title, image?, url?, id?}
```

**Comment by mydearxym** [comment]
is there still a plan to support mentions feature?

**Comment by zsoflin** [comment]
Yes, is this still planned? Unfortunately, mentions are necessary in order for me to switch from Quill :)

**Comment by SammyVimes** [comment]
Can you share the designs of mentions plugin? I want to implement it (I need it for my pet project and I'd be happy to share implementation with community)

**Comment by neSpecc** [comment]
@SammyVimes here is a draft:

 ![](https://capella.pics/a2c03ea9-0998-4069-85b5-2549fddd53a6.jpg)

Try to use Inline Tools API. If you will need some methods that we does not provide, write us.

**Comment by SammyVimes** [comment]
@neSpecc I guess mentions tool should listen for every input change (in case of new character is `@`) unlike other tools. I hope that will not affect performance too much  😃

**Comment by Steve1820** [comment]
@SammyVimes Did you end up implemented this?

Let me know if I can help.

**Comment by ckissi** [comment]
@SammyVimes @Steve1820  I'm curious too. It would be great to have it.

**Comment by GuruM** [comment]
@SammyVimes @Steve1820 @ckissi yeah I'm going to attempt this now but would love to know if anyone has been able to do it. I saw https://github.com/mydearxym/editor-mention but I couldn't get that working.

**Comment by akshayrathodar** [comment]
> Yes, is this still planned? Unfortunately, mentions are necessary in order for me to switch from Quill :)

i created the plugin for same for my own use

**Comment by aromero86** [comment]
@akshayrathodar this plugin is public? thanks

**Comment by akshayrathodar** [comment]
>  this plugin is public? thanks

no but i can help you regard this or can share you my code

**Reply by Ruubber** [reply]
@akshayrathodar
Hello.
I need to implement the mention function. Can you please share the code?
Thank you!

**Comment by aromero86** [comment]
@akshayrathodar it would be nice have an starting point. I would be very grateful if you could share your code. Of course, if something interesting comes from there, I would go to the EditorJS community

> no but i can help you regard this or can share you my code

**Comment by romainhe** [comment]
@akshayrathodar Hi, can you share your code ? I've the same needs. Thx

**Comment by kennyd98** [comment]
@akshayrathodar Hi, I will also appreciate if you share me the code. Thank you.

**Reply by akshayrathodar** [reply]
https://github.com/akshayrathodar/editor-js-audio

**Comment by jeremyVignelles** [comment]
If I understand this issue properly, the workflow would be:
1. Type the name of the person you want to mention
2. Click on the inline tool
3. The mention is now linked.

I don't find this workflow optimal, and I think we need a new kind of tool to tackle this issue, so I created #1623

**Comment by russelshane** [comment]
This might be the one you're looking for, although, it needs a backend that has a search functionality, should be easy if you already have a backend API running:

@mrpritchett/editorjs-mentions

**Comment by Fennec-hub** [comment]
@neSpecc

I think I can create a plugin for handling suggestions like `mentions`, `hashtags` and `emojis` ..., they all follow more or less the same logic, I looked into the `Inline Tools API`, it doesn't  provide any method to listen to  text change or any kind of `keyup` event, there is only the `checkstate` method that triggers only when a selection is made.

If you can give some insight on how to achieve the following, I'll be glad to start working on this plugin and share the result.

1. **A way to detect text input changes ideally `onkeyup` event**
2. **Get the current word at the caret position ideally with a `Range` object**

And the workflow should be:

1. Detect text change (`keyup` event listener) 
2. Get the current word at the caret position with a Range object
4. Check for some starting key characters (`@` for mention, `#` for hashtags, `:` for emojis ...)
5. Call an `API` endpoint for a list of suggestions
6. Display a drop down menu with the suggestions list
7. Replace the current word with the selected suggestion and the surrounding tags (<a>, <span> ....);

I tried a work around by getting the current focused block element from `api.blocks` and add a `keyup` listener but it's hacky and not viable solution.

**Reply by icastillejogomez** [reply]
Some updates here? I need this implementation I help if needed

**Reply by geekyayush** [reply]
Hey man, have you created such plugin?

**Comment by akshayrathodar** [comment]
Sure, will do that. actually need to check that project which is in my old
laptop. will check . please ping me on +919429076228 (whatsapp) . if i will
forgot.

**Reply by Ruubber** [reply]
@akshayrathodar 
Thank you!!
How can I share the code with you?
Can I contact you on whats app?

**Reply by akshayrathodar** [reply]
https://github.com/akshayrathodar/editor-js-audio Please check some module i hv created

**Comment by samandar-boymurodov** [comment]
Hi, do you know if this feature has been implemented?

**Comment by davidpawar** [comment]
I would need a similar feature. I need to have replacer elements, which the user can place in certain location inside his text.

e.g.

Writing a reply mail.

Dear [Contact Person],
we will send you the items to [Destination]

**Comment by bettysteger** [comment]
I hope i can help someone with this: https://gist.github.com/bettysteger/d8f660030849ae85211d51d93e736a3b

**My solution** of getting this to work:

1. Listening to paragraphs for inputting `@`
2. Replace `@` with `<a rel="tag">@</a>` and select the new anchor tag to trigger the custom mention inline tool
3. Mention inline tool opens up an user list, where you can search and select the user to tag!

Inline output will be `<a href="#${user.id}" rel="tag">${user.name}</a>`.

 Feedback is highly appreciated!

**Comment by Hesesses** [comment]
Hello,

Is there any timeline for this feature? We are consider editorjs for our upcoming project

**Reply by akshayrathodar** [reply]
Check this repo. may be it will help you. https://github.com/akshayrathodar/editor-js-audio

**Reply by akshayrathodar** [reply]
Here you can check zip for @editor.js
